core/lib/Thelia/Action/Product.php

```diff
@@ -110,7 +112,14 @@ public function update(ProductUpdateEvent $event)
                     ->save($con)
                 ;
 
-                // Update default category (ifd required)
+                // Update default PSE (if product has no attributes and the product's ref change)
+                if ($prev_ref != $product->getRef() && $product->countSaleElements() == 1) {
```

Comment:
This checks if there is 1 PSE, not if there are no attributes
What if there is 1 attribute combination ? :-)
You should add a condition to your test if $product->getDefaultSaleElements()->countAttributeCombinations() == 0